src/constructs/cloudwatch/alarm.ts

```diff
@@ -2,13 +2,27 @@ import { Alarm } from "@aws-cdk/aws-cloudwatch";
 import type { AlarmProps } from "@aws-cdk/aws-cloudwatch";
 import { SnsAction } from "@aws-cdk/aws-cloudwatch-actions";
 import { Topic } from "@aws-cdk/aws-sns";
-import type { ITopic } from "@aws-cdk/aws-sns";
 import { Stage } from "../../constants";
 import type { GuStack } from "../core";
 import type { AppIdentity } from "../core/identity";
+import { GuAlarmTopicParameter } from "../core/parameters/sns";
 
 export interface GuAlarmProps extends AlarmProps, AppIdentity {
-  snsTopicName: string;
+  /**
+   * Topic ARN.
+   *
+   * @default account alarm topic - see {@link GuAlarmTopicParameter}.".
+   */
+  snsTopicArn?: string;
+
+  /**
+   * Topic Name.
+   *
+   * @deprecated prefer {@link GuAlarmProps.snsTopicArn} instead.
+   * @default account alarm topic - see {@link GuAlarmTopicParameter}.
+   */
+  snsTopicName?: string;
```

Comment:
I think the reason that I originally preferred name to `ARN` was because it allows teams to encode this directly in the template (i.e. it avoids using CFN parameters, which often need to be populated manually) without leaking private information (like account number).

Reply:
Thanks, that makes sense. Perhaps we shouldn't deprecate this field but support both.